### PR TITLE
Fixes #1611: Adding comments

### LINKED ISSUE #1611: Add chance to add comments to Jobs
Right now there is no chance to add any comment to the jobs results. It would be cool to add this opportunity to enable people to collaborate better directly inside the platform

### PR DESCRIPTION
**Pull request by 0x0elliot**
# Description

Basic comment support.

## Type of change

- [x] New feature (non-breaking change which adds functionality).

# Checklist

- [x] I have read and understood the rules about [how to Contribute](https://intelowl.readthedocs.io/en/latest/Contribute.html) to this project
- [x] The pull request is for the branch `develop`
- [x] Linters (Black, Flake, Isort) gave 0 errors. If you have correctly installed [pre-commit](https://intelowl.readthedocs.io/en/latest/Contribute.html#how-to-start-setup-project-and-development-instance), it does these checks and adjustments on your behalf.

# Screenshot

Screenshot 2023-04-02 at 5 37 07 AM

### CODE REVIEW

**Comment by 0x0elliot** [comment]
god, these pre-commit errors always mess up for me. that's why i would love #1626 to work out.

**Comment by 0x0elliot** [comment]
there is pagination that i am yet to add. i would rather do it another weekend. could only take out this saturday for intelowl :)

**Comment by 0x0elliot** [comment]
Should be ready to go!

**Comment by mlodic** [comment]
> there is pagination and comment searching that i am yet to add. i would rather do it another weekend. want me to open another issue for it?

yep that would be cool!

Meanwhile thank you for this PR! That is really cool feature! :)

In the future, if you are willing to try out to add also few tests from the frontend part it would be great! We recently added the framework to add those tests too (https://intelowl.readthedocs.io/en/latest/Contribute.html#id2). That would increase stability of the application!

Plus, feel free to create branches directly inside IntelOwl main repository instead of your branch. That would help us other people to collaborate because ,for instance, it would be simpler to add their changes on top of your opened PRs

File: api_app/serializers.py
Comment (mlodic) [inline api_app/serializers.py:313]: this triggers:

```
  File "/opt/deploy/intel_owl/api_app/serializers.py", line 313, in validate
    if not job.user.memberships.filter(organization=user.organization).exists():
AttributeError: 'User' object has no attribute 'memberships
```

**Review by mlodic — CHANGES_REQUESTED** [review]
The concept is cool! Aside the error that I found, it seems a cool feature to add!

File: api_app/views.py
Comment (0x0elliot) [reply api_app/views.py:470]: what? no only comment owner can.

File: api_app/serializers.py
Comment (0x0elliot) [reply api_app/serializers.py:323]: okay noted

File: frontend/src/components/jobs/result/utils/CommentOverview.jsx
Comment (0x0elliot) [reply frontend/src/components/jobs/result/utils/CommentOverview.jsx:56]: yes sure.

File: api_app/views.py
Comment (mlodic) [reply api_app/views.py:470]: mine was only a suggestion to change the comment, I should have been more explicit in that sense.

**Comment by mlodic** [comment]
hey @0x0elliot, we are about to push [major changes ](https://github.com/intelowlproject/IntelOwl/pull/1625) to the `develop` branch (we will do it tomorrow). To reduce the chance of conflicts with this PR, I would like to merge this PR before the bigger one. Would you be able to make those fixes today? Otherwise if I can do I'll do it today. Thank you

**Comment by 0x0elliot** [comment]
okay i will

File: api_app/serializers.py
Comment (0x0elliot) [reply api_app/serializers.py:313]: let me fix

File: api_app/views.py
Comment (0x0elliot) [reply api_app/views.py]: is it fine if we leave it the way it is? i will edit it in the next PR if you want me to. i want to finish this issue safely before we get into the visualizer hassle.

Comment (0x0elliot) [reply api_app/views.py]: i am not sure if i like this logic right now. i think only organization admins should be able to do all that rather than anyone in the organization, to begin with. this is why i would like to wait out the change.

**Comment by 0x0elliot** [comment]
should be good to go. i don't want this to be a hassle we need to look into right now. @mlodic 

i will make all the small changes we need in a separate PR.

**Comment by 0ssigeno** [comment]
I have made some fixes here and there. Before merging this PR, there are some tests missing. 
I have made the boilerplate inside `tests.api_app.test_views.CommentViewSetTestCase` and `tests.api_app.test_serializers.CommentSerializerTestCase`.

**Comment by 0x0elliot** [comment]
Ready?

**Comment by 0ssigeno** [comment]
@drosetti is fixing some stuff on the frontend if i'm not mistaken. I will merge this PR once it is done.
Meanwhile, if you have more tests or you want to refactor stuff that I have changed, you are free to do it!

**Comment by 0x0elliot** [comment]
interesting. let me know if you need help for this @drosetti

**Comment by drosetti** [comment]
Don't worry I fixed it. :)

I used the `vh` instead of `px` as units for the height: this solved the problem of the oldest messages and also makes the UI more adaptable to different devices. Pixels number can be a problem with different displays: in 1920x1080 500 px takes some space in 4k the UI is completely different. vh instead is based on the height of the viewport.

Also i sorted the messages in this way the most recent are near the comment input and are shown immediately.

**Pull request merged**
